src/robot/libdocpkg/xmlwriter.py

```diff
@@ -15,33 +15,50 @@
 
 from robot.utils import XmlWriter, get_timestamp
 
+from .htmlwriter import DocToHtml
+
 
 class LibdocXmlWriter(object):
 
+    def __init__(self, html_doc_format=False):
+        self._html_doc_format = html_doc_format
+
     def write(self, libdoc, outfile):
+        formatter = DocToHtml(libdoc.doc_format) \
+            if self._html_doc_format \
+            else lambda doc: doc
         writer = XmlWriter(outfile)
-        writer.start('keywordspec', {'name': libdoc.name, 'type': libdoc.type,
-                                     'format': libdoc.doc_format,
-                                     'generated': get_timestamp(millissep=None)})
+        self._write_start(writer, libdoc, formatter)
+        self._write_keywords(writer, 'init', libdoc.inits, formatter)
+        self._write_keywords(writer, 'kw', libdoc.keywords, formatter)
+        self._write_end(writer)
+
+    def _write_start(self, writer, libdoc, formatter):
+        doc_format = 'HTML' if self._html_doc_format else libdoc.doc_format
+        lib_attrs = {'name': libdoc.name,
+                     'type': libdoc.type,
+                     'format': doc_format,
+                     'generated': get_timestamp(millissep=None)}
+        writer.start('keywordspec', lib_attrs)
         writer.element('version', libdoc.version)
         writer.element('scope', libdoc.scope)
         writer.element('namedargs', 'yes' if libdoc.named_args else 'no')
-        writer.element('doc', libdoc.doc)
-        self._write_keywords('init', libdoc.inits, writer)
-        self._write_keywords('kw', libdoc.keywords, writer)
-        writer.end('keywordspec')
-        writer.close()
+        writer.element('doc', formatter(libdoc.doc))
 
-    def _write_keywords(self, type, keywords, writer):
+    def _write_keywords(self, writer, type, keywords, formatter):
```

Comment:
why isn't `writer` an instance attribute?

Reply:
That's actually how @Snooz82 implemented it in the beginning. :smile:

Reply:
When coding object oriented i like to use instance attributes over handing over and returning stuff. As Pekka said, its kind of a personal style.

Because in Python the `self.` decreases the readability i have no real preferences here.